packages/analytics-core/src/diagnostics/diagnostic.ts

```diff
@@ -0,0 +1,74 @@
+import { Diagnostic as IDiagnostic, DiagnosticOptions } from '@amplitude/analytics-types';
+import { DIAGNOSTIC_ENDPOINT } from '../constants';
+import { DiagnosticEvent } from './typings';
+import { DIAGNOSTIC_METADATA_TYPE } from './constants';
+
+export class Diagnostic implements IDiagnostic {
+  isDisabled = false;
+  serverUrl: string = DIAGNOSTIC_ENDPOINT;
+  apiKey?: string = '';
+  queue: DiagnosticEvent[] = [];
+
+  private scheduled: ReturnType<typeof setTimeout> | null = null;
+  // deault delay is 1 minute
+  // make it private to prevent users from changing it to smaller value
+  private delay = 60000;
+
+  constructor(options?: DiagnosticOptions) {
+    this.isDisabled = options && options.isDisabled ? options.isDisabled : false;
+    this.serverUrl = options && options.serverUrl ? options.serverUrl : DIAGNOSTIC_ENDPOINT;
+    if (options && options.apiKey) {
+      this.apiKey = options.apiKey;
+    }
+  }
+
+  track(eventCount: number, code: number, message: string) {
+    if (this.isDisabled) {
+      return;
+    }
+
+    this.queue.push(this.diagnosticEventBuilder(eventCount, code, message));
+
+    if (!this.scheduled) {
+      this.scheduled = setTimeout(() => {
+        void this.flush();
+      }, this.delay);
+    }
+  }
+
+  async flush(): Promise<void> {
+    // send http request based on environment
+    // implemented in its child class
+
```

Comment:
Maybe add a default implemetion here? Log the diagnostics to loggerProvider? See other comment above.

Reply:
leave it as how it is now but we will see how we can improve it as Node and RN have diagnosticProvdier